feat: json persistence nanodb

ErezSha opened the pull request:
implements a json based persistence for nanodb

closes #51
closes #85

erez-sharim committed:
feat: json persistence nanodb

implements a json based persistence for nanodb
moves the locking mechanism from the store level (UserStore, AgentStore) to the DB level

resolves #51

erez-sharim committed:
feat: mongodb implementation

adds a mongodb database implementation

resolves #85

asaf commented:
Hey, this works well!

Did minor changes to settings, for extensibility and best practice should use a single var `DB_URI` for DB configuration using URIs:

```
FLUX0_DB_URI=nanodb://memory
FLUX0_DB_URI=nanodb://json?dir=./data
FLUX0_DB_URI=mongodb://localhost:27017
```

asaf committed:
fix: refactor db settings to a single env var using URI

For extensibility and best practice, modified DB settings via a single var FLUX0_DB_URI.

Examples:

FLUX0_DB_URI=nanodb://memory
FLUX0_DB_URI=nanodb://json?dir=./data
FLUX0_DB_URI=mongodb://localhost:27017
